src/main/java/net/minecraftforge/resource/ResourceCacheManager.java

```diff
@@ -69,35 +89,39 @@
      * @param indexOffThreadConfig True to index the file-tree off-thread.
      * @param pathBuilder          The path builder to use.
      */
+    @Deprecated(since = "1.19.2", forRemoval = true)
     public ResourceCacheManager(final boolean supportsReloading, final ForgeConfigSpec.BooleanValue indexOffThreadConfig, final BiFunction<PackType, String, Path> pathBuilder)
     {
         this.supportsReloading = supportsReloading;
-        this.indexOffThreadConfigOption = indexOffThreadConfig;
+        this.indexOnThreadConfigurationKey = Iterators.getLast(indexOffThreadConfig.getPath().iterator());
         this.pathBuilder = pathBuilder;
     }
 
-    public boolean shouldIndexOffThread()
+    public ResourceCacheManager(final boolean supportsReloading, final String indexOnThreadConfigurationKey, final BiFunction<PackType, String, Path> pathBuilder)
     {
-        return !getConfigValue(this.indexOffThreadConfigOption);
+        this.supportsReloading = supportsReloading;
+        this.indexOnThreadConfigurationKey = indexOnThreadConfigurationKey;
+        this.pathBuilder = pathBuilder;
     }
 
+    /**
+     * Indicates if the caching system is enabled or not.
+     *
+     * @return {@code true} if the caching system is enabled, {@code false} otherwise.
+     */
```

Comment:
```suggestion
    /**
     * {@return if the caching system is enabled}
     */
```

Reply:
There is a significant difference between the returns and the method body documentation in that it describes the returned values.